frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -141,20 +142,34 @@ function buildFieldFilterUiParameter(
   );
   const mappedFields = mappingsForParameter.map(mapping => {
     const { target, card } = mapping;
-    return getTargetFieldFromCard(target, card, metadata);
+    const question = new Question(card, metadata);
+    const field = getTargetFieldFromCard(target, card, metadata);
+
+    return { field, shouldResolveFkField: !question.isNative() };
   });
-  const fields = mappedFields.filter((field): field is Field => field != null);
+
   const hasVariableTemplateTagTarget = mappingsForParameter.some(mapping => {
     return isVariableTarget(mapping.target);
   });
-  const uniqueFieldsWithFKResolved = _.uniq(
-    fields.map(field => field.target ?? field),
+
+  const uniqueFields = _.uniq(
+    mappedFields
+      .filter(
+        (
+          mappedField,
+        ): mappedField is { field: Field; shouldResolveFkField: boolean } => {
+          return mappedField.field != null;
+        },
+      )
+      .map(({ field, shouldResolveFkField }) => {
+        return shouldResolveFkField ? field.target ?? field : field;
```

Comment:
This makes sure that we only resolve the foreign key filters on structured questions. Because we should match a dashboard filter mapped to a native question behavior with the native question parameter behavior.

In this issue, we tested that both native question filter and dashboard filters mapped to the same native question got the same filter widget and loaded the filter values the same way.

By allowing resolving foreign key fields we make sure we use the correct remapping when the resolved foreign key field supports it e.g. in the following screenshot, a dashboard ID filter is mapped to the `Orders.User_ID` field (question created using query builder)
![image](https://github.com/metabase/metabase/assets/1937582/e7a2b4cd-19b6-4c58-a176-c5b844b0d006)

There are also E2E tests covering such a case
https://github.com/metabase/metabase/blob/e84bd827709a60dd069385144f25f653eaa93e98/e2e/test/scenarios/dashboard-filters/dashboard-filters-id.cy.spec.js#L61-L72